README.md

```diff
@@ -84,149 +94,106 @@ const UserProfile = ({ userId }) => (
-import { useResourceState } from "react-remote-resource";
+type Load<A> = (...args: Array<any>) => A | Promise<A>;
 
-const ProductCategory = ({ categoryId }) => {
-  const [categoriesById] = useResourceState(categoriesResource);
-  const [products] = useResourceState(productsResource, [categoryId]);
+type Loader<A> = (currentState: ?A, refresh: boolean) => Load<A>;
 
-  return (
-    <section>
-      <header>
-        <h1>{categoriesById[categoryId].name}</h1>
-      </header>
-      <ul>
-        {products.map(product => (
-          <li key={product.id}>
-            <Link to={`/products/${product.id}`}>{product.name}</Link>
-          </li>
-        ))}
-      </ul>
-    </section>
-  );
+type Resource<A> = {
+  // The generated UUID for the resource
+  id: string,
+  // A function that takes the current state and a refresh flag and returns a function that takes any arguments and returns the next state or a Promise that resolves with the next state
+  refresh: Load<A>,
+  // Returns the current state of the resource
+  getState: () => A,
+  // A function that takes the next state or a function that receives the current state and returns the next state.
+  setState: (A | A => A) => void,
+  // Allows for subscribing to resource state changes. Basically a wrapper around store.subscribe.
+  subscribe: (() => void) => void,
+  // A react hook that allows you to use a resource entry's state in the same way that you would use React's useState
+  useEntryState: <A>(...args: Array<any>) => [A, (A | A => A) => void]
 };
 ```
 
-This hook is very powerful. Let's walk through what happens when it is used:
+### `createSingleEntryResource`
 
-1. If there is no data for this resource OR no entry in the cache for the arguments (See [invalidateAfter](https://github.com/chadwatson/react-remote-resource/blob/master/README.md#createremoteresource)) OR the cache is stale (See [invalidateAfter](https://github.com/chadwatson/react-remote-resource/blob/master/README.md#createremoteresource)), then the `load` function of the `categoriesResource` and `productsResource` will be invoked and promises thrown.
-2. If either of the promises reject, the closest `RemoteResourceBoundary` will handle the error. If both promises resolve, the `categoriesById` and `products` data will be available to use (as the first item in the tuple).
-3. You can set the state using the second item in the tuple. Resource state changes, unlike component based `useState`, will persist in memory. If a component unmounts and remounts the state will be the same as when you left it.
+An opinionated version of `createResource` that assumes there is only one entry in the resource state. The getter function simply returns the last data that was fetched. The setter function simply sets the resource state to the data that was fetched. And the predicate function simply checks if the resource state is not `undefined`. Basically the loader function will only be called once if it successfully resolves.
```

Comment:
I think this makes more sense. But what does `(pre set)` mean?

Reply:
Got ya! It means that the value is pre set for the user. Mainly an indicator to let the user know that the values are pre-applied for them when the resource is created.

Thoughts?

Reply:
Gotcha. I think there may be a better way to communicate that. Really I think if we just removed the `#### loader` section it would get the job done. The sentence before it says `It takes a loader function and creates a resource using the following inputs`, which suggests that the following things will be pre-filled for you. Or maybe it would be better to say something like:

```markdown
Creates a resource that only has one entry. It conveniently supplies getter, setter, and predicate functions to `createResource` under the hood, allowing you to simply supply a function that fetches your data. Once your data is fetched it will not be refetched.
```

Then maybe it would be okay to leave the details of the getter, setter, and predicate out. Thoughts?

Reply:
Love it!